embassy-stm32l0/src/exti.rs

```diff
@@ -51,6 +54,47 @@ impl<T: PinWithInterrupt + 'static> ExtiPin<T> {
     }
 }
 
+impl<T: InputPin + PinWithInterrupt + 'static> ExtiPin<T> {
+    fn wait_for_state<'a>(self: Pin<&'a mut Self>, state: bool) -> impl Future<Output = ()> + 'a {
+        let line = self.pin.line();
+        let s = unsafe { self.get_unchecked_mut() };
+
+        Exti::unpend(line);
```

Comment:
the InterruptFuture already does unpend of the interupt, I think this is not needed.

Reply:
This unpends the exti line, not the NVIC.

Reply:
see https://github.com/embassy-rs/embassy/blob/master/embassy-stm32f4/src/exti.rs#L108

Reply:
oh whoops my bad :)